spec/core/matchers/matchersUtilSpec.js

```diff
@@ -93,6 +93,24 @@ describe("matchersUtil", function() {
       expect(jasmineUnderTest.matchersUtil.equals(new Error("foo"), new Error("bar"))).toBe(false);
     });
 
+    it("fails for objects with different constructors", function() {
+      function One() {}
+      function Two() {}
+
+      expect(jasmineUnderTest.matchersUtil.equals(new One(), new Two())).toBe(false);
+    });
+
+    if (typeof document === 'object') {
+      it("passes for equivalent objects from different frames", function() {
```

Comment:
This spec is testing the same behavior as these: https://github.com/benchristel/jasmine/blob/5f27c1f199a9ffe57869e7e9fc99b9abddfca508/spec/core/matchers/matchersUtilSpec.js#L209-L234

I'm cool with also testing this in browsers, but can you move this down with the others (maybe in a shared describe). 

Also we should unify on a testing style for specs that only execute in certain environments.